tests/test_zmq_ppoll_signals.cpp

```diff
@@ -0,0 +1,241 @@
+        send_string_expect_success(socket, "breaker breaker", 0);
+        recv_string_expect_success(socket, "one-niner", 0);
+
+        // then send the signal
+        kill(child_pid, SIGTERM);
+
+        // for good measure, and to make sure everything went as expected, close off with another handshake, which will trigger the second poll call on the other side
+        send_string_expect_success(socket, "breaker breaker", 0);
+        recv_string_expect_success(socket, "one-niner", 0);
+
+        // finish
+        test_context_socket_close (socket);
+
+        // wait for child
+        int status = 0;
+        pid_t pid;
+        do {
+            pid = waitpid(child_pid, &status, 0);
+        } while (-1 == pid && EINTR == errno); // retry on interrupted system call
+
+        if (0 != status) {
+            if (WIFEXITED(status)) {
+                printf("exited, status=%d\n", WEXITSTATUS(status));
+            } else if (WIFSIGNALED(status)) {
+                printf("killed by signal %d\n", WTERMSIG(status));
+            } else if (WIFSTOPPED(status)) {
+                printf("stopped by signal %d\n", WSTOPSIG(status));
+            } else if (WIFCONTINUED(status)) {
+                printf("continued\n");
+            }
+        }
+
+        if (-1 == pid) {
+            printf("waitpid returned -1, with errno %s\n", strerror(errno));
+        }
+    } else { // child
+        setup_test_context ();
+        // set up signal mask and install handler for SIGTERM
+        struct sigaction sa;
+        memset(&sa, '\0', sizeof(sa));
+        sa.sa_handler = handle_sigterm;
+        TEST_ASSERT_SUCCESS_ERRNO (sigaction(SIGTERM, &sa, NULL) );
+
+        void *socket = test_context_socket (ZMQ_REP);
+        TEST_ASSERT_SUCCESS_ERRNO (zmq_connect(socket, "tcp://127.0.0.1:6660"));
+
+        zmq_pollitem_t pollitems[] = {
+          {socket, 0, ZMQ_POLLIN, 0},
+          };
+
+        // first receive test message and send back handshake
+        recv_string_expect_success(socket, "breaker breaker", 0);
+        send_string_expect_success(socket, "one-niner", 0);
+
+        // now start ppolling, which should exit with EINTR because of the SIGTERM
+        TEST_ASSERT_FAILURE_ERRNO(EINTR, zmq_poll(pollitems, 1, -1));
+        TEST_ASSERT_TRUE(sigterm_received);
+
+        // poll again for the final handshake
+        TEST_ASSERT_SUCCESS_ERRNO(zmq_poll(pollitems, 1, -1));
+        TEST_ASSERT_BITS_HIGH (ZMQ_POLLIN, pollitems[0].revents);
+        // receive and send back handshake
+        recv_string_expect_success(socket, "breaker breaker", 0);
+        send_string_expect_success(socket, "one-niner", 0);
+
+        // finish
+        sleep (1); // wait before closing socket, so that parent has time to receive
+        test_context_socket_close (socket);
+        _Exit(0);
+    }
+}
+*/
+
+int main ()
+{
+    UNITY_BEGIN ();
+    RUN_TEST (test_ppoll_signals);
+//    RUN_TEST (test_poll_signals);
```

Comment:
don't add commented out stuff

Reply:
It was meant as an illustration for devs of why it is necessary. I will remove it.